interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/FragmentComponentDefinitionExtractor.scala

```diff
@@ -101,37 +117,66 @@ class FragmentComponentDefinitionExtractor(
 
   private def toParameter(
       componentConfig: SingleComponentConfig
-  )(p: FragmentParameter): Writer[List[FragmentParamClassLoadErrorData], Parameter] = {
-    val paramName = p.name
-    p.typ
+  )(fragmentParameter: FragmentParameter): Writer[List[FragmentParamClassLoadErrorData], Parameter] = {
+    fragmentParameter.typ
       .toRuntimeClass(classLoader)
       .map(Typed(_))
       .map(Writer.value[List[FragmentParamClassLoadErrorData], TypingResult])
       .getOrElse(
         Writer
           .value[List[FragmentParamClassLoadErrorData], TypingResult](Unknown)
-          .tell(List(FragmentParamClassLoadErrorData(paramName, p.typ.refClazzName)))
+          .tell(List(FragmentParamClassLoadErrorData(fragmentParameter.name, fragmentParameter.typ.refClazzName)))
       )
-      .map(toParameter(componentConfig, paramName, _))
+      .map(toParameter(componentConfig, _, fragmentParameter))
   }
 
-  private def toParameter(componentConfig: SingleComponentConfig, paramName: String, typ: typing.TypingResult) = {
-    val config          = componentConfig.params.flatMap(_.get(paramName)).getOrElse(ParameterConfig.empty)
-    val parameterData   = ParameterData(typ, Nil)
-    val extractedEditor = EditorExtractor.extract(parameterData, config)
+  private def toParameter(
+      componentConfig: SingleComponentConfig,
+      typ: typing.TypingResult,
+      fragmentParameter: FragmentParameter
+  ) = {
+    val config        = componentConfig.params.flatMap(_.get(fragmentParameter.name)).getOrElse(ParameterConfig.empty)
+    val parameterData = ParameterData(typ, Nil)
+
+    val extractedEditor = fragmentParameter match {
+      case param: FragmentParameterFixedValuesUserDefinedList =>
+        fixedValuesEditorWithInputMode(
+          param.inputMode,
+          FixedValuesParameterEditor(
+            param.fixedValuesList.map(v => FixedExpressionValue(v.expression, v.label))
+          )
+        )
+      case _: FragmentParameterNoFixedValues => EditorExtractor.extract(parameterData, config)
+    }
+
+    val isOptional = !fragmentParameter.required
     Parameter
-      .optional(paramName, typ)
+      .optional(fragmentParameter.name, typ)
       .copy(
         editor = extractedEditor,
         validators = ValidatorsExtractor
-          .extract(ValidatorExtractorParameters(parameterData, isOptional = true, config, extractedEditor)),
-        // TODO: ability to pick a default value from gui
```

Comment:
Nice, fixing an old TODO :)

Reply:
same happened in BaseFlowTest.scala, it is a nice feeling